kubeadm/v1.15.0/KubeCluster.ps1

```diff
@@ -0,0 +1,321 @@
+            Plugin = @{
+                Name = "vxlan";
+            };
+            InterfaceName = "Ethernet";
+        }
+    }
+
+    if ($Global:ClusterConfiguration.Cni.Plugin.Name -eq "vxlan")
+    {
+        if (!$Global:ClusterConfiguration.Kubernetes.KubeProxy)
+        {
+            $Global:ClusterConfiguration.Kubernetes | Add-Member -MemberType NoteProperty -Name KubeProxy -Value @{
+                    Gates = "WinOverlay=true";
+            }
+        }
+    }
+
+    if (!$Global:ClusterConfiguration.Cri)
+    {
+        $Global:ClusterConfiguration | Add-Member -MemberType NoteProperty -Name Cri -Value @{
+            Name = "dockerd";
+            Images = @{
+                Pause = "mcr.microsoft.com/k8s/core/pause:1.0.0";
+                Nanoserver = "mcr.microsoft.com/windows/nanoserver:1809";
+                ServerCore = "mcr.microsoft.com/windows/servercore:ltsc2019";
+            }
+        }
+    }
+}
+
+###############################################################################################
+# Download pre-req scripts
+
+Import-Module "$PSScriptRoot\KubeClusterHelper.psm1"
+$hnsPath = "https://raw.githubusercontent.com/Microsoft/SDN/master/Kubernetes/windows/hns.psm1"
+$hnsDestination = "$PSScriptRoot\hns.psm1" 
+DownloadFile -Url $hnsPath -Destination $hnsDestination
+Import-Module $hnsDestination
+
+
+ReadKubeclusterConfig -ConfigFile $ConfigFile
+InitHelper
+PrintConfig
+WriteKubeClusterConfig
+
+
+# Initialize internal network modes of windows corresponding to 
+# the plugin used in the cluster
+$Global:NetworkName = "cbr0"
+$Global:NetworkMode = "l2bridge"
+if ($Global:NetworkPlugin -eq "vxlan")
+{
+    $Global:NetworkMode = "overlay"
+    $Global:NetworkName = "vxlan0"
+}
+######################################################################################################################
+
+# Handle --install
+function
+Restart-And-Run()
+{
+    Write-Output "Restart is required; restarting now..."
+
+    $argList = $script:MyInvocation.Line.replace($script:MyInvocation.InvocationName, "")
+
+    #
+    # Update .\ to the invocation directory for the bootstrap
+    #
+    $scriptPath = $script:MyInvocation.MyCommand.Path
+
+    $argList = $argList -replace "\.\\", "$pwd\"
+
+    if ((Split-Path -Parent -Path $scriptPath) -ne $pwd)
+    {
+        $sourceScriptPath = $scriptPath
+        $scriptPath = "$pwd\$($script:MyInvocation.MyCommand.Name)"
+
+        Copy-Item $sourceScriptPath $scriptPath
+    }
+
+    Write-Output "Creating scheduled task action ($scriptPath $argList)..."
+    $action = New-ScheduledTaskAction -Execute "powershell.exe" -Argument "-NoExit $scriptPath $argList"
+
+    Write-Output "Creating scheduled task trigger..."
+    $trigger = New-ScheduledTaskTrigger -AtLogOn
+
+    Write-Output "Registering script to re-run at next user logon..."
+    Register-ScheduledTask -TaskName $global:BootstrapTask -Action $action -Trigger $trigger -RunLevel Highest | Out-Null
```

Comment:
This is failing for me. Where is $global:BootstrapTask set?

Reply:
Resolved. Missed a change.